pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -455,8 +467,35 @@ func (lbaas *CloudLb) ensureLoadBalancerNodes(lbID uint64, port corev1.ServicePo
 		}
 	}
 
+	// Delete obsolete nodes for this pool
+	for _, node := range memberNodes {
+		klog.V(4).Infof("Deleting obsolete node %d for loadbalancer %s address %s", node.ID, lbID, node.Address)
+		err := lbnodes.Delete(lbaas.lb, lbID, node.ID).ExtractErr()
+		if err != nil && !cpoerrors.IsNotFound(err) {
+			return fmt.Errorf("error deleting obsolete node %d for load balancer %d address %s: %v", node.ID, lbID, node.Address, err)
+		}
+		provisioningStatus, err := waitLoadbalancerActiveStatus(lbaas.lb, lbID)
+		if err != nil {
+			return fmt.Errorf("timeout when waiting for loadbalancer to be ACTIVE after deleting member, current provisioning status %s", provisioningStatus)
+		}
+	}
+
+	// Re-fetch current members after deletion
+	memberNodes, err = getNodesByLBID(lbaas.lb, lbID)
+	if err != nil && !cpoerrors.IsNotFound(err) {
+		return fmt.Errorf("error getting load balancer nodes after deletion %d: %v", lbID, err)
+	}
+
+	allowedToAdd := MaxNodeLimit - len(memberNodes)
+	if allowedToAdd < 0 {
+		allowedToAdd = 0
+	}
+	if allowedToAdd < len(addNodes) {
+		addNodes = addNodes[:allowedToAdd]
+	}
```

Comment:
Either of these if conditions don't seem like normal situation to occur. Can we at least log so that we know at a later point if we are hitting this and may need to investigate why.